tools/src/test/java/org/apache/kafka/tools/FeatureCommandTest.java

```diff
@@ -212,10 +212,10 @@ public void testHandleUpgrade() {
         namespace.put("dry_run", false);
         String upgradeOutput = ToolsTestUtils.captureStandardOut(() -> {
             Throwable t = assertThrows(TerseException.class, () -> FeatureCommand.handleUpgrade(new Namespace(namespace), buildAdminClient()));
-            assertTrue(t.getMessage().contains("1 out of 2 operation(s) failed."));
+            assertTrue(t.getMessage().contains("2 out of 2 operation(s) failed."));
         });
-        assertEquals(format("foo.bar was upgraded to 6.%n" +
-            "Could not upgrade metadata.version to 5. Can't upgrade to lower version."), upgradeOutput);
+        assertEquals(format("Could not upgrade foo.bar to 6. Invalid update version 5 for feature metadata.version. Can't upgrade to lower version.%n" +
```

Comment:
A couple of questions on the error message.
1. I thought we now fail fast and the error message will just contain the first failed feature. Why does the error message contain both features?
2. Why is "Invalid update version 5 for feature metadata.version. Can't upgrade to lower version" repeated?

Reply:
This is how the UpdateFeaturesResult looks (and how it looked in the last PR). The RPC call contains just one message, but the result will contain an error for every feature in the request. This is consistent with the previous behavior for top level errors. I don't think this should be changed. 

The question is whether you want to change the behavior of FeatureCommand to not output all the features and just the single error response message.

Reply:
Thanks for the explanation. We can leave it as it is for now.